nsis/src/nsis/tests/NsisExecutableTest.java

```diff
@@ -98,4 +97,23 @@ public void testNsisCreationBzipCompressed() throws IOException, InvalidFormatEx
 		}
 	}
 
+	/**
+	 * Deobfuscated the inputStream object passed as a parameter. The obfuscation is
+	 * a simple XOR with 0x55
+	 * 
+	 * @param inputStream
+	 * @return the deobfuscated ByteArrayProvider object
+	 * @throws IOException
+	 */
+	private ByteArrayProvider deobfuscate(InputStream inputStream) throws IOException {
```

Comment:
Since this is test and we know the size of the inputs, I see no problem with this implementation.

Had this code been outside of a test I would suggest using a `FilterInputStream` https://docs.oracle.com/javase/7/docs/api/java/io/FilterInputStream.html#available()

In your subclass you would overwrite all the read methods and do the xor in those methods.

Keep this in mind in case we need to do something similar elsewhere.

Reply:
Yeah I had this reflection about using a ByteArrayProvider and since we know that we are using relatively small test files, I came to the same conclusion. I'll keep the FilterInputStream in mind